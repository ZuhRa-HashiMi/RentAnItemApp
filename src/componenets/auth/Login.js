import React, { Component } from 'react';
import axios from 'axios';
import PropTypes from 'prop-types';

export default class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
    };

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }

  handleSubmit(event) {
    const { email, password } = this.state;
    const { handleSuccessfulAuth } = this.props;

    axios
      .post(
        'http://localhost:3000/sessions',
        {
          user: {
            email,
            password,
          },
        },
        { withCredentials: true },
      )
      .then((response) => {
        if (response.data.logged_in) {
          handleSuccessfulAuth(response.data);
        }
      })
      .catch((error) => {
        console.log('login error', error);
      });
    event.preventDefault();
  }

  render() {
    const { email, password } = this.state;
    return (
      <div>
        <form onSubmit={this.handleSubmit}>
          <input
            type="email"
            name="email"
            placeholder="Email"
            value={email}
            onChange={this.handleChange}
            required
          />

          <input
            type="password"
            name="password"
            placeholder="Password"
            value={password}
            onChange={this.handleChange}
            required
          />

          <button type="submit">Login</button>
        </form>
      </div>
    );
  }
}

Login.propTypes = {
  handleSuccessfulAuth: PropTypes.func.isRequired,
};
