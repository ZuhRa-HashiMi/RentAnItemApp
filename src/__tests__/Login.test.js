import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../componenets/auth/Login';

it('renders Footer correctly', () => {
  const tree = renderer.create(<Login />);
  expect(tree).toMatchSnapshot();
});
