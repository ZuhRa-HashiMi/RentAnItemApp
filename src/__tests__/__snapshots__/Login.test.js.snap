// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer correctly 1`] = `
<div
  className="add-house-section"
>
  <div
    className="backdrop column flex-center"
  >
    <h1>
      Sign in
    </h1>
    <p>
      Hello there! Sign in and start managing your system
    </p>
    <form
      className="column"
      onSubmit={[Function]}
    >
      <input
        className="input-text"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
      <input
        className="input-text"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
      <p />
      <button
        className="btn"
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
